Fix delete actor bug

PStahl opened the pull request:
When deleting an actor, it is not removed from the application's list of actors in the storage.

To reproduce:
csruntime --host localhost --port 5001 --controlport 5002 --keep-alive -l INFO
cscontrol http://localhost:5002 deploy calvin/examples/sample-scripts/actions.calvin
cscontrol http://localhost:5002 actor delete <actor_id>
csweb
Then, if you select the application in the web interface, you'll get an error because the actor id is still in the application's list of actors, so it will try to fetch that actor, but since it no longer exists, it'll fail.

HaraldGustafsson commented:
Currently the app is static when it comes to actors and connections, which could be changed as you suggest. But note that the storage chosen for application assumes static actors. Your changes will end in race condition if several runtimes makes changes to the application. The proper change of this is to store the set of actors for an application separately e.g. with prefix "app-actors-" and using append, remove and get_concat storage methods. This change should not affect the storage api for the application or actor, i.e. add/get/delete_application and add/delete_actor, they just also maintain the set of actor for an application. Could you break out these changes in a PR separate from replication etc.

PStahl commented:
Change to use storage append and remove "app-actors-" methods. Therefore also added a new function for getting application actors, also updated the web interface to use this.

It would be nice to change though so that only one request is needed from the web interface, e.g. getApplication instead of loadApplicationData and loadApplicationActors.

Not completely sure about the case when actors are deleted during migration. Do you mean that the delete request takes place when the migration is in place, before it creates it on the new node? in this case, what is the desired behaviour?

olaan commented:
Parts of this request have been considered and (optionally) included during rewrites and updates. Thanks!
